### DISCUSSION TITLE
Force close browse gallery picker images

### DISCUSSION THREAD

**Discussion by levanhoangvu** [yasirkula/UnityNativeGallery#293] [Q&A] [answered]
Hi, could we force close browse gallery picker images in code unity?

**Comment by yasirkula** [comment, marked as answer]
I think the native iOS source code ([NativeGallery.mm](https://github.com/yasirkula/UnityNativeGallery/blob/master/Plugins/NativeGallery/iOS/NativeGallery.mm)) could be modified to achieve this:

To the top:

```objc
+ (void)closeMediaPicker;
```

To the middle:

```objc
+ (void)closeMediaPicker
{
	if( imagePicker != nil )
		[imagePicker dismissViewControllerAnimated:NO completion:nil];
#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
	else if( CHECK_IOS_VERSION( @"14.0" ) && imagePickerNew != nil )
		[imagePickerNew dismissViewControllerAnimated:NO completion:nil];
#endif
}
```

To the bottom:

```objc
extern "C" void _NativeGallery_CloseMediaPicker()
{
	[UNativeGallery closeMediaPicker];
}
```

In C#:

```csharp
#if !UNITY_EDITOR && UNITY_IOS
[System.Runtime.InteropServices.DllImport( "__Internal" )]
private static extern void _NativeGallery_CloseMediaPicker();
#endif

void X()
{
	_NativeGallery_CloseMediaPicker();
}
```

**Reply by levanhoangvu** [reply]
Ohh, thanks you for quick answer. Also android is it possible too?

**Reply by yasirkula** [reply]
Android should pause Unity activity while the image picker is displayed? Was it not the case for you?

**Reply by levanhoangvu** [reply]
Ah, i see @yasirkula . You saved my life. Many thanks!!
